omod/src/test/java/org/openmrs/module/idgen/web/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,137 @@
+    assertEquals(autoGenerationOption.getId(), PropertyUtils.getProperty(result, "id"));
+    assertEquals(autoGenerationOption.getIdentifierType(), PropertyUtils.getProperty(result, "identifierType"));
+    assertEquals(autoGenerationOption.getLocation(), PropertyUtils.getProperty(result, "location"));
+    assertEquals(autoGenerationOption.getSource(), PropertyUtils.getProperty(result, "source"));
+    assertEquals(autoGenerationOption.isAutomaticGenerationEnabled(), PropertyUtils.getProperty(result, "isAutomaticGenerationEnabled"));
+    assertEquals(autoGenerationOption.isManualEntryEnabled(), PropertyUtils.getProperty(result, "isManualEntryEnabled"));
+  }
+
+  @Test
+  public void shouldGetAnAutoGenerationOptionByPatientIdentifierType() throws Exception {
+    MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + ptid);
+    SimpleObject result = deserialize(handle(req));
+    
+    AutoGenerationOption autoGenerationOption = service.getAutoGenerationOption(ptid);
+    assertEquals(autoGenerationOption.getId(), PropertyUtils.getProperty(result, "id"));
+    assertEquals(autoGenerationOption.getIdentifierType(), PropertyUtils.getProperty(result, "identifierType"));
+    assertEquals(autoGenerationOption.getLocation(), PropertyUtils.getProperty(result, "location"));
+    assertEquals(autoGenerationOption.getSource(), PropertyUtils.getProperty(result, "source"));
+    assertEquals(autoGenerationOption.isAutomaticGenerationEnabled(), PropertyUtils.getProperty(result, "isAutomaticGenerationEnabled"));
+    assertEquals(autoGenerationOption.isManualEntryEnabled(), PropertyUtils.getProperty(result, "isManualEntryEnabled"));
+  }
+
+  @Test
+  public void shouldCreateNewAutoGenerationOption() throws Exception {
+    long originalCount = getAllCount();
+
+    SimpleObject autoGenerationOption = new SimpleObject();
+    autoGenerationOption.add("identifierType", "idgen");
+    autoGenerationOption.add("location", "National Hospital");
+    autoGenerationOption.add("source", "New source");
+    autoGenerationOption.add("isAutomaticGenerationEnabled", true);
+    autoGenerationOption.add("isManualEntryEnabled", true);
+
+    String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+
+    MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+    req.setContent(json.getBytes());
+    
+    SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "id"));
+		assertEquals(originalCount + 1, getAllCount());
+  }
+
+  @Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+    final String newLocation = "Updated Location";
+    final String newSource = "Updated Source";
+		SimpleObject autoGenerationOption = new SimpleObject();
+    autoGenerationOption.add("location", newLocation);
+    autoGenerationOption.add("source", newSource);
+		
```

Comment:
Can you create a test dataset file and edit an existing one?

Reply:
I obviously don't know how to do that. Can you please point me to the right resources that will help?

Reply:
Did you first ask from your team mates?

Reply:
I asked and I tried the option provided by @adamsdenniskariuki but it didn't work as well.

Reply:
Can i see what you tried?

Reply:
He said I should replace what I have in my `doGetAll` method with this ` return Context.getService(IdentifierSourceService.class).getAutoGenerationOptions(Context.getPatientService().getPatientIdentifierType(2))`. I did and it's still the same.

Reply:
Am not asking about doGetAll. Do you see that this review comment is in a file different from the one that has doGetAll?